server/src/main/java/com/cloud/configuration/ConfigurationManagerImpl.java

```diff
@@ -3332,14 +3312,14 @@ public DiskOffering updateDiskOffering(final UpdateDiskOfferingCmd cmd) {
     }
 
     /**
-     * Check the tags parameters to the diskOffering
+     * Check the tags parameters to the disk/service offering
      * <ul>
      *     <li>If tags is null, do nothing and return.</li>
-     *     <li>If tags is not null, set tag to the diskOffering.</li>
-     *     <li>If tags is an blank string, set null on diskOffering tag.</li>
+     *     <li>If tags is not null, set tag to the disk/service offering.</li>
+     *     <li>If tags is an blank string, set null on disk/service offering tag.</li>
      * </ul>
      */
-    protected void updateDiskOfferingTagsIfIsNotNull(String tags, DiskOfferingVO diskOffering) {
+    protected void updateOfferingTagsIfIsNotNull(String tags, DiskOfferingVO diskOffering) {
         if (tags == null) { return; }
         if (StringUtils.isNotBlank(tags)) {
             diskOffering.setTags(tags);
```

Comment:
I think it would be interesting to use the same logic as the hostTags conditional here, allowing you to remove existing tags.

Reply:
@SadiJr, I'm not sure for which logic you are talking about. This method allows the removal of tags. Unfortunately, I don't have time these days to check on this PR and to add the required changes.

Reply:
@slavkap for example, suppose you have the tags "tag1,tag2" saved and you want to remove them. Ideally an empty string ("") would be sent, which would remove the tags in the database in a similar way to the host tag treatment in the updateServiceOfferingHostTagsIfNotNull method. However, this type of scenario is not covered here.

Reply:
@SadiJr, both methods are absolutely the same, with one difference that this one updates the storage tags and updateServiceOfferingHostTagsIfNotNull updates the host tags. And with both, you can add/remove tags

     protected void updateOfferingTagsIfIsNotNull(String tags, DiskOfferingVO diskOffering) {
        if (tags == null) { return; }
        if (StringUtils.isNotBlank(tags)) {
            diskOffering.setTags(tags);
        } else {
            diskOffering.setTags(null);
        }
    }

    /**
     * Check the host tags parameters to the service offering
     * <ul>
     *     <li>If host tags is null, do nothing and return.</li>
     *     <li>If host tags is not null, set host tag to the service offering.</li>
     *     <li>If host tags is an blank string, set null on service offering tag.</li>
     * </ul>
     */
    protected void updateServiceOfferingHostTagsIfNotNull(String hostTags, ServiceOfferingVO offering) {
        if (hostTags == null) {
            return;
        }
        if (StringUtils.isNotBlank(hostTags)) {
            offering.setHostTag(hostTags);
        } else {
            offering.setHostTag(null);
        }
    }
 
 
set the storage tags:

http://localhost:5050/client/api/?id=86a7d7bd-42a1-4d78-be8f-51a13f3eef7c&amp;name=Small Instance&amp;displaytext=Small Instance&amp;tags=tag1, tag2&amp;command=updateServiceOffering&amp;response=json

response:


```
 {
 	"updateserviceofferingresponse": {
 		"serviceoffering": {
 			"id": "86a7d7bd-42a1-4d78-be8f-51a13f3eef7c",
 			"name": "Small Instance",
 			"displaytext": "Small Instance",
 			"cpunumber": 1,
 			"cpuspeed": 500,
 			"memory": 512,
 			"created": "2021-02-26T07:38:24+0200",
 			"storagetype": "shared",
 			"provisioningtype": "thin",
 			"offerha": false,
 			"limitcpuuse": false,
 			"isvolatile": false,
 			"tags": "tag1, tag2",
 			"hosttags": "",
 			"issystem": false,
 			"defaultuse": false,
 			"iscustomized": false,
 			"rootdisksize": 0
 		}
 	}
 }
```


remove the storage tags:

http://localhost:5050/client/api/?id=86a7d7bd-42a1-4d78-be8f-51a13f3eef7c&name=Small Instance&displaytext=Small Instance&tags=&command=updateServiceOffering&response=json

   and the response:
   

```
    {
 	"updateserviceofferingresponse": {
 		"serviceoffering": {
 			"id": "86a7d7bd-42a1-4d78-be8f-51a13f3eef7c",
 			"name": "Small Instance",
 			"displaytext": "Small Instance",
 			"cpunumber": 1,
 			"cpuspeed": 500,
 			"memory": 512,
 			"created": "2021-02-26T07:38:24+0200",
 			"storagetype": "shared",
 			"provisioningtype": "thin",
 			"offerha": false,
 			"limitcpuuse": false,
 			"isvolatile": false,
 			"hosttags": "",
 			"issystem": false,
 			"defaultuse": false,
 			"iscustomized": false,
 			"rootdisksize": 0
 		}
 	}
 }
```

Reply:
@slavkap Ok, my bad. I reviewed the code and saw that I was making a mistake. I apologize for any inconvenience.

Reply:
@SadiJr, it's Ok, we cleared up the misunderstanding :)